File: samples/client/3_0_3_unit_test/java/src/main/java/org/openapijsonschematools/client/RootServerInfo.java
package org.openapijsonschematools.client;

import org.openapijsonschematools.client.servers.RootServer0;
import org.openapijsonschematools.client.servers.Server;
import org.openapijsonschematools.client.servers.ServerProvider;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.Objects;

public class RootServerInfo {
    public static class RootServerInfo1 implements ServerProvider<ServerIndex> {
        private final RootServer0 server0;

        RootServerInfo1(
            @Nullable RootServer0 server0
        ) {
            this.server0 = Objects.requireNonNullElseGet(server0, RootServer0::new);
        }

        @Override
        public Server getServer(ServerIndex serverIndex) {
            return server0;
        }
    }

    public static class RootServerInfoBuilder {
        private @Nullable RootServer0 server0;

        public RootServerInfoBuilder() {}

        public RootServerInfoBuilder rootServer0(RootServer0 server0) {
            this.server0 = server0;
            return this;
        }

        public RootServerInfo1 build() {
            return new RootServerInfo1(
                server0
            );
        }
    }

    public enum ServerIndex {
        SERVER_0
    }
}

File: samples/client/3_1_0_unit_test/kotlin/src/main/kotlin/org/openapijsonschematools/client/RootServerInfo.kt
package org.openapijsonschematools.client

import org.openapijsonschematools.client.servers.RootServer0
import org.openapijsonschematools.client.servers.Server
import org.openapijsonschematools.client.servers.ServerProvider

class RootServerInfo {
    class RootServerInfo1(
        private val server0: RootServer0 = RootServer0()
    ) : ServerProvider<ServerIndex> {

        override fun getServer(serverIndex: ServerIndex): Server {
            return server0
        }
    }

    enum class ServerIndex {
        SERVER_0
    }
}